engine/src/main/scala/cromwell/engine/workflow/tokens/JobTokenDispenserActor.scala

```diff
@@ -125,11 +138,34 @@ class JobTokenDispenserActor(override val serviceRegistryActor: ActorRef,
       ()
     }
 
-  private def dispense(n: Int) = if (tokenQueues.nonEmpty) {
+  private def checkAndDispenseTokens(n: Int): Unit =
+    if (tokenQueues.nonEmpty) {
+      // don't fetch cloud quota exhausted groups for token dispenser allocating 'restart' tokens
+      if (dispenserType == "execution" && groupMetricsActor.nonEmpty) {
+        groupMetricsActor.get
```

Comment:
We try to avoid unprotected `get`s - could use a case statement instead here, like 
```
(dispenserType, groupMetricsActor) match {
    case ("execution", Some(a)) => a.ask...
    case _ => dispense(...)
}
```

Reply:
I did add `groupMetricsActor.nonEmpty` on L44 check before doing the `.get`. Would you still recommend to use pattern matching here instead of what I have?

Reply:
Yeah, it's true that the current code state is guaranteed to not cause a problem, but the compiler isn't checking that, and it's easy for the association between the `nonEmpty` check and the `.get` to degrade over time as the code is changed.

Reply:
I see your point. I will change it not use `.get` and refactor the way you suggested 👍